Scripts/UtilityScripts/AzTSUtility.ps1

```diff
@@ -0,0 +1,1617 @@
+  .SYNOPSIS
+  This command removes a Key Vault network access rule for the specified public IP address.
+  .DESCRIPTION
+  This command removes a Key Vault network access rule for the specified public IP address.
+  .PARAMETER SubscriptionId
+  The Azure subscription ID containing the Key Vault.
+  .PARAMETER ResourceGroupName
+  The Resource Group name containing the Key Vault.
+  .PARAMETER KeyVaultName
+  The Key Vault name.
+  .PARAMETER PublicIpAddress
+  The public IP address to match to the rule to be removed. Must be in CIDR format; for a single IP address, put it in form a.b.c.d/32
+  .INPUTS
+  None
+  .OUTPUTS
+  None
+  .EXAMPLE
+  PS> Remove-KeyVaultNetworkAccessRuleForIpAddress -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -KeyVaultName "MyKeyVaultName" -PublicIpAddress "1.1.1.1"
+  .LINK
+  None
+#>
+
+[CmdletBinding()]
+param (
+  [Parameter(Mandatory = $true)]
+  [string]
+  $SubscriptionId,
+  [Parameter(Mandatory = $true)]
+  [string]
+  $ResourceGroupName,
+  [Parameter(Mandatory = $true)]
+  [string]
+  $KeyVaultName,
+  [Parameter(Mandatory = $true)]
+  [string]
+  $PublicIpAddress
+)
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  Write-Information -InformationAction Continue -MessageData "Removing rule for $PublicIpAddress"
+
+  Remove-AzKeyVaultNetworkRule `
+    -SubscriptionId $SubscriptionId `
+    -ResourceGroupName $ResourceGroupName `
+    -VaultName $KeyVaultName `
+    -IpAddressRange $PublicIpAddress
+}
+# ####################################################################################################
+
+# ####################################################################################################
+# Azure_Subscription_AuthZ_Remove_Deprecated_Accounts
+
+function Get-DeletedUser()
+{
+  <#
+    .SYNOPSIS
+    This command shows a deleted user.
+    .DESCRIPTION
+    This command shows a deleted user. This is helpful when trying to translate a scanner status message user GUID to a human display name in order to locate the user on a list of assignments.
+    This command requires the Microsoft.Graph SDK. See installation instructions: https://learn.microsoft.com/powershell/microsoftgraph/installation?view=graph-powershell-1.0#installation
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the Key Vault.
+    .PARAMETER DeletedUserId
+    The user GUID for the deleted user.
+    .INPUTS
+    None
+    .OUTPUTS
+    None
+    .EXAMPLE
+    PS> Get-DeletedUser -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -DeletedUserId "00000000-xxxx-0000-xxxx-000000000000"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory=$true)]
+    [string]
+    $DeletedObjectId
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  Connect-MgGraph
+
+  Get-MgDirectoryDeletedItem -DirectoryObjectId $DeletedObjectId
```

Comment:
Is this command part of Az module or module 'Microsoft.Graph.Identity.DirectoryManagement' need to be separately installed?

Reply:
The function description notes "This command requires the Microsoft.Graph SDK. See installation instructions: https://learn.microsoft.com/powershell/microsoftgraph/installation?view=graph-powershell-1.0#installation" but I will also add this to the README for this function.